process/block/preprocess/sovereignChainTransactions.go

```diff
@@ -214,3 +220,89 @@ func (sct *sovereignChainTransactions) isTransactionEligibleForExecution(tx *tra
 func isCriticalError(err error) bool {
 	return err != nil && !errors.Is(err, process.ErrHigherNonceInTransaction)
 }
+
+// RequestTransactionsForMiniBlock requests missing transactions for the given mini block
+func (sct *sovereignChainTransactions) RequestTransactionsForMiniBlock(miniBlock *block.MiniBlock) int {
+	if miniBlock == nil {
+		return 0
+	}
+
+	//TODO: Implement functionality for sovereign chain
```

Comment:
I do not think this has to be reimplemented - it can use the base RequestTransactionsForMiniblock and request information from p2p on his ownsovereign system.

Reply:
Right, I've deleted it so the base one can be used